### PR TITLE
RPP Tensor Audio Support - Slice

### PR DESCRIPTION
**Pull request by r-abishek**
This PR modifies the already added voxel slice kernels to now accept anchor and shape params for a more generic version for Audio support.

Slice can now function in all 3 test suites:
- In the audio test suite by calling it with a 1D float tensor.
- In the image test suite by calling it with 2D or 3D tensors (1 channel / 3 channel images)
- In the voxel test suite by calling it with a 4D tensor. (3 channel 3D voxel images)

The PR adds support to fill the padded region with fill value when padding is enabled. It also modifies the non-silent region kernel output types to make it compatible with latest slice.
All support added for QA/performance tests in image/audio/voxel test suites.

### CODE REVIEW

**Comment by kiritigowda** [comment]
@r-abishek - please resolve merge conflicts

**Comment by LakshmiKumar23** [comment]
@r-abishek can you resolve merge conflicts?

**Comment by r-abishek** [comment]
@kiritigowda @LakshmiKumar23 All merge conflicts resolved and CI passing

**Comment by kiritigowda** [comment]
@r-abishek please resolve conflicts.

File: src/modules/cpu/kernel/slice.hpp [hunk 1]
```diff
@@ -26,184 +26,213 @@ SOFTWARE.
...
+                    srcPtrDepth = srcPtrChannel;
+                    dstPtrDepth = dstPtrChannel;
+                    for(int i = 0; i < maxDepth; i++)
+                    {
+                        T *srcPtrRow, *dstPtrRow;
+                        srcPtrRow = srcPtrDepth;
+                        dstPtrRow = dstPtrDepth;
+                        for(int j = 0; j < maxHeight; j++)
+                        {
+                            memcpy(dstPtrRow, srcPtrRow, copyLengthInBytes);
```
Comment (rrawther) [inline src/modules/cpu/kernel/slice.hpp:108]: Is this as fast as doing in AVX?

**Review by rrawther — CHANGES_REQUESTED** [review]
Please check if we can write a fast mem_cpy using AVX

**Comment by rrawther** [comment]
@AryanSalmanpour : Can you please review the HIP kernels in this PR?

**Review by rrawther — APPROVED** [review]

**Review by AryanSalmanpour — APPROVED** [review]

**Comment by kiritigowda** [comment]
@r-abishek -- needs merge conflicts resolved

**Pull request merged**